dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskDefinitionServiceImpl.java

```diff
@@ -820,9 +820,6 @@ public Map<String, Object> updateTaskWithUpstream(User loginUser, long projectCo
         Map<String, Object> result = new HashMap<>();
         TaskDefinitionLog taskDefinitionToUpdate =
                 updateTask(loginUser, projectCode, taskCode, taskDefinitionJsonObj, result);
-        if (result.get(Constants.STATUS) != Status.SUCCESS && taskDefinitionToUpdate == null) {
-            return result;
-        }
```

Comment:
This is used to ensure that meaningless multiple modifications will not be directly written to the database. We should add the judgement of the predecessor node instead of remove it.

Reply:
I took a quick look at this part of the code and found that you are right. But if we remove this, each modification will be saved to the database regardless of whether there is a real change. It makes the judgment of equals meaningless.

I'm +0 to this way. What do you guys think? @ruanwenjun @caishunfeng @EricGao888 @zhongjiajie

Reply:
> Why invalid modifications affect the database

Each modification will be insert to the `t_ds_task_definition_log` as a newer version.

https://github.com/apache/dolphinscheduler/blob/8439b5dc69f2a7dfc1043393bbbb21a8277715da/dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskDefinitionServiceImpl.java#L754-L775

Reply:
Looks good to me.
But I found that some useless code, can you delete together?
https://github.com/apache/dolphinscheduler/blob/8be32d4145c851a01d7300cd5a854be9b4a27055/dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskDefinitionServiceImpl.java#L839-L841